gradient_utils/metrics/tensorboard/patch.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import logging
+from importlib import import_module, reload
+from gradient_utils import metrics
+
+logger = logging.getLogger(__name__)
+TENSORBOARD_C_MODULE = "tensorboard.python.ops.gen_summary_ops"
+TENSORBOARD_WRITER_MODULE = "tensorboard.summary.writer.event_file_writer"
+TENSORBOARDX_WRITER_MODULE = "tensorboardX.event_file_writer"
+TENSORBOARD_PYTORCH_MODULE = "torch.utils.tensorboard.writer"
+
+
+def patch():
+    if len(metrics.patched["tensorboard"]) > 0:
+        raise ValueError(
+            "Tensorboard already patched. You may be calling metrics.init() more than once."
+        )
+
+    # Make sure tensorboard is installed
+    _get_module("tensorboard", required=True)
+
+    # Grab specific tensorboard modules for patching
+    c = _get_module(TENSORBOARD_C_MODULE)
+    tb = _get_module(TENSORBOARD_WRITER_MODULE)
+    tbx = _get_module(TENSORBOARDX_WRITER_MODULE)
+    pt = _get_module(TENSORBOARD_PYTORCH_MODULE)
+
+    if not c and not tb and not pt:
```

Comment:
What's a valid config? I'm not familiar so hard to say if the and's should be or's or if all 4 need to be required.

Reply:
basically, if none of these are defined, raise an exception. i'll update the error to make more sense.